### PR TITLE
add lib/*.jar to classpath of plugin-jetty7, plugin-jetty8, plugin-jetty9, like plugin-jetty93

### PR DESCRIPTION
**Pull request by AngiWANG**
the classpath of plugin-jetty93 contain lib/*.jar, but not in plugin-jetty7, plugin-jetty8, plugin-jetty9, otherwise when import rjr into Eclipse, it will prompt classpath error.

ps: plugin-jetty93 based on jetty 9.3.\* which required JDK 8, i create plugin-jetty92 based on jetty 9.2.\* which required JDK 7, i have tested it, works well, could i contribute it?

### CODE REVIEW

**Comment by xzer** [comment]
In fact, it is a mistake that lib/* is contained in plugin-jetty93's class path.

I am not sure why you got classpath error, in my environment, there is no error, and even I removed the lib/* from 93's classpath, there is no error too.

I guess you import the source into a normal eclipse installation rather the PDE version which contains the necessary plugins for plugin development, which plugins will assign all the lib/* into classpath by respecting the extra plugin configuration files.

Anyway, thanks very much for your pull request that suggests the potential wrong configuration, I will push my fix that removes the lib/* from 93's classpath. 

Finally, you are very welcome to join the development of this plugin, so, please let me know if you have any problem to build the development environment.

**Comment by xzer** [comment]
Further, 9.2 version support is very nice, you can push it by another PR.

**Comment by AngiWANG** [comment]
oh, i see.
I have fetched your recent commit "remove rebundant classpath..." and tested as a result, only if the eclipse installed **Tycho Configurator** and the file **MANIFEST.MF** contains libs in the **Bundle-ClassPath**, it's almost ture as you said "assign '_**all the lib/***_' into classpath". plugin-jetty93 is ok below, but plugin-jetty7, plugin-jetty8 and plugin-jetty9 aren't.
```
...
Bundle-ClassPath: .,
 lib/annotations/asm-5.0.1.jar,
 lib/annotations/asm-commons-5.0.1.jar,
 lib/annotations/javax.annotation-api-1.2.jar,
 lib/apache-jsp/org.eclipse.jdt.core.compiler.ecj-4.4.2.jar,
 lib/apache-jsp/org.eclipse.jetty.apache-jsp-9.3.6.v20151106.jar,
 lib/apache-jsp/org.mortbay.jasper.apache-el-8.0.27.jar,
 lib/apache-jsp/org.mortbay.jasper.apache-jsp-8.0.27.jar,
 lib/apache-jstl/org.apache.taglibs.taglibs-standard-impl-1.2.5.jar,
 ...
```
Another point, **bootstrap** is not a source folder, so maybe it's a reason why no classpath error, maybe i could contribute it.

**Comment by xzer** [comment]
which eclipse version are you using? The bootstrap folder is extactly configured as source folder.

I am using Version: Mars.1 Release (4.5.1) Build id: 20150924-1200 for current development.

I guess I know what is the reason...

The following is what I installed for the plugin development:

>   Apache Jakarta log4j Source Bundle	1.2.15.v201012070815	org.apache.log4j.source	Eclipse Orbit
  AspectJ Compiler	1.8.7.20150910082200	org.aspectj.feature.group	Eclipse AspectJ Development Tools
  AspectJ Compiler Source Code	1.8.7.20150910082200	org.aspectj.source.feature.group	Eclipse AspectJ Development Tools
  AspectJ Development Tools	2.2.4.e45x-20151204-1000	org.eclipse.ajdt.feature.group	Eclipse AspectJ Development Tools
  AspectJ Development Tools (AJDT) Source Code	2.2.4.e45x-20151204-1000	org.eclipse.ajdt.source.feature.group	Eclipse AspectJ Development Tools
  Cross References tool (XRef)	2.2.4.e45x-20151204-1000	org.eclipse.contribution.xref.feature.group	Eclipse AspectJ Development Tools
  Cross References Tool Source Code	2.2.4.e45x-20151204-1000	org.eclipse.contribution.xref.source.feature.group	Eclipse AspectJ Development Tools
  Easy Shell Feature	1.4.2	com.tetrade.eclipse.plugins.easyshell.feature.feature.group	Marcel Schoen, Andre Bossert
  Eclipse IDE for Eclipse Committers	4.5.1.20150917-1200	epp.package.committers	null
  Eclipse Weaving Service Feature	2.2.4.e45x-20151204-1000	org.eclipse.contribution.weaving.feature.group	Eclipse AspectJ Development Tools
  Eclipse Weaving Service Source Code	2.2.4.e45x-20151204-1000	org.eclipse.contribution.weaving.source.feature.group	Eclipse AspectJ Development Tools
  Eclipse Web Developer Tools	3.7.1.v201509022125	org.eclipse.wst.web_ui.feature.feature.group	Eclipse Web Tools Platform
  Equinox Weaving SDK	1.1.0.weaving-hook-20140821	org.eclipse.equinox.weaving.sdk.feature.group	Eclipse Equinox Project
  Hamcrest Library of Additional Matchers	1.3.0.v201305281000	org.hamcrest.library	Eclipse Orbit
  Hamcrest Library of Additional Matchers Source	1.3.0.v201305281000	org.hamcrest.library.source	Eclipse Orbit
  m2e - Maven Integration for Eclipse (includes Incubating components)	1.6.2.20150902-0002	org.eclipse.m2e.feature.feature.group	Eclipse.org - m2e
  m2e - slf4j over logback logging (Optional)	1.6.2.20150902-0002	org.eclipse.m2e.logback.feature.feature.group	Eclipse.org - m2e
  SWTBot for Eclipse Forms Testing	2.3.0.201506081302	org.eclipse.swtbot.forms.feature.group	Eclipse.org
  SWTBot for Eclipse Testing	2.3.0.201506081302	org.eclipse.swtbot.eclipse.feature.group	Eclipse.org
  SWTBot for GEF Testing	2.3.0.201506081302	org.eclipse.swtbot.eclipse.gef.feature.group	Eclipse.org
  SWTBot for SWT Testing	2.3.0.201506081302	org.eclipse.swtbot.feature.group	Eclipse.org
  SWTBot IDE Features	2.3.0.201506081302	org.eclipse.swtbot.ide.feature.group	Eclipse.org
  SWTBot JUnit Headless launchers for Eclipse	2.3.0.201506081302	org.eclipse.swtbot.eclipse.test.junit.feature.group	Eclipse.org
  SWTBot Recorder and Test Generator (Provisional)	2.3.0.201506081302	org.eclipse.swtbot.generator.feature.feature.group	Eclipse.org


The above is hard to read due to the line breaking, copy it to other text editors to read.

I am sorry there is lack of the necessary dependency installation description because I also knew little about it when the first time I took over the ownership of this project, thus I forgot to record what I did for the environment.

Basically, you need install all the above plugins for developing this plugin, and it would be perfect if you can create a description about how to build the development environment.

**Comment by xzer** [comment]
The following is where I installed the dependencies from, but there are duplicated items in different version, I have forgotten which I used, please confirm it to the versions I suggested above.

> AJDT Update Site	http://download.eclipse.org/tools/ajdt/42/update	Enabled
AJDT Update Site	http://download.eclipse.org/tools/ajdt/43/update	Enabled
ajdt-e45x-20150610-1700	http://download.eclipse.org/tools/ajdt/45/dev/update	Enabled
EasyShell Release Versions	http://anb0s.github.io/EasyShell	Enabled
m2e releases repository	http://download.eclipse.org/technology/m2e/releases	Enabled
m2e-wtp Juno Release repository	http://download.eclipse.org/m2e-wtp/releases/juno/	Enabled
Mars	http://download.eclipse.org/releases/mars/	Enabled
Maven Integration for Eclipse Update Site	http://download.eclipse.org/technology/m2e/releases/1.5/1.5.0.20140606-0033	Enabled
org.eclipse.e4.core.tools.update	http://download.vogella.com/kepler/e4tools	Enabled
org.eclipse.swtbot.updatesite	http://download.eclipse.org/technology/swtbot/releases/latest/	Enabled
The Eclipse Project Updates	http://download.eclipse.org/eclipse/updates/4.5	Enabled
The Eclipse Web Tools Platform (WTP) software repository	http://download.eclipse.org/webtools/repository/mars/	Enabled
Update Site	http://run-jetty-run.googlecode.com/svn/trunk/updatesite/	Enabled

**Comment by AngiWANG** [comment]
**Thanks for your work.**
I am using Eclipse for RCP and RAP Developers, Version: Mars.2 Release (4.5.2) Build id: 20160218-0600. and only installed Tycho Confiturator.

I kown your classpath have configured bootstrap as a source folder, but when maven -> update project, something changed.
before:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
	<classpathentry kind="src" path="src"/>
	<classpathentry kind="src" path="bootstrap"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
	<classpathentry kind="output" path="bin"/>
</classpath>
```
after:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
	<classpathentry kind="src" path="src/"/>
	<classpathentry exported="true" kind="lib" path="lib/annotations/asm-5.0.1.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/annotations/asm-commons-5.0.1.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/annotations/javax.annotation-api-1.2.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jsp/org.eclipse.jdt.core.compiler.ecj-4.4.2.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jsp/org.eclipse.jetty.apache-jsp-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jsp/org.mortbay.jasper.apache-el-8.0.27.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jsp/org.mortbay.jasper.apache-jsp-8.0.27.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jstl/org.apache.taglibs.taglibs-standard-impl-1.2.5.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/apache-jstl/org.apache.taglibs.taglibs-standard-spec-1.2.5.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/fcgi/fcgi-client-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/fcgi/fcgi-server-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/http2/http2-common-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/http2/http2-hpack-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/http2/http2-server-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jaspi/javax.security.auth.message-1.0.0.v201108011116.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jndi/javax.mail.glassfish-1.4.1.v201005082020.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jndi/javax.transaction-api-1.2.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/monitor/jetty-monitor-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/setuid/jetty-setuid-java-1.0.3.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/spring/jetty-spring-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/javax-websocket-client-impl-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/javax-websocket-server-impl-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/javax.websocket-api-1.0.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/websocket-api-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/websocket-client-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/websocket-common-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/websocket-server-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/websocket/websocket-servlet-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/cdi-core-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/cdi-servlet-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/cdi-websocket-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/gcloud-session-manager-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-alpn-server-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-annotations-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-client-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-continuation-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-deploy-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-http-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-infinispan-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-io-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-jaas-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-jaspi-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-jmx-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-jndi-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-nosql-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-plus-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-proxy-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-quickstart-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-rewrite-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-schemas-3.1.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-security-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-server-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-servlet-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-servlets-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-util-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-webapp-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/jetty-xml-9.3.6.v20151106.jar"/>
	<classpathentry exported="true" kind="lib" path="lib/servlet-api-3.1.jar"/>
	<classpathentry kind="output" path="target/classes"/>
</classpath>
```
**<classpathentry kind="src" path="bootstrap"/>** is removed and something else is changed. i think maven makes when building Eclipse plug-ins with maven.

**Comment by xzer** [comment]
It is strange that why you CAN use maven in eclipse, the maven configuration is only used to package in command line and I did not add maven facet into the eclipse project definition, which is not necessary.

I installed m2e but I found that I am not using it.  All the dependencies are managed by eclipse plugin mechanism.

**Comment by AngiWANG** [comment]
i have done some tests, but it's strange also.
could you show me your local .classpath and .project of plugin-jetty9 and plugin-jetty93?

**Comment by AngiWANG** [comment]
Maybe i have found the reason. Assign project 's classpath by respecting Bundle-ClassPath. so i think Bundle-ClassPath should be set for plugin-jetty7, plugin-jetty8, plugin-jetty9, like plugin-jetty93.

**Comment by xzer** [comment]
At first, I pushed all the local files to the repository, so there is no difference between my local .classpath/.project and the ones in repository. Also, there is no difference between the j7's .classpath and j93's .classpath file except different jdk version.

Secondly, yes, I did not use Tycho for eclipse building, the tycho is only used with maven to package the plugin in command line.  Also, the maven is not considered as the part of eclipse building. The whole project is managed via the classical eclipse feature/plugin mechanism rather than maven/tycho.

Finally, I do not know what different between the eclipse PDE version and RCP version, but I think it may be helpful to try the PDE version rather than RCP version.

**Comment by AngiWANG** [comment]
Doesn't your local .classpath contains lib/*.jar?

**Comment by xzer** [comment]
No, it doesnt, just as the same as the pushed file.
